model/riscv_insts_zce_ext.sail

```diff
@@ -0,0 +1,1021 @@
+                         RETIRE_SUCCESS
+                       },
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
+}
+
+mapping clause assembly = C_LBU(uimm, rsc, rdc)
+  <-> "c.lbu" ^ spc() ^ creg_name(rdc) ^ sep() ^ creg_name(rsc) ^ sep() ^ hex_bits_5(uimm @ 0b000)
+/* ****************************************************************** */
+union clause ast = C_LHU : (bits(1), cregidx, cregidx)
+
+//$[wavedrom FUNCT3 _ _ _ _ _ C0]
+mapping clause encdec_compressed = C_LHU(ui5, rs1, rd)
+  <-> 0b100 @ 0b001 @ rs1 : cregidx @ 0b0 @ ui5 : bits(1) @ rd : cregidx @ 0b00
+
+function clause execute (C_LHU(uimm, rsc, rdc)) = {
+  let immext : xlenbits = EXTZ(uimm);
+  let imm : bits(12) = EXTZ(uimm);
+  
+  let rd = creg2reg_idx(rdc);
+  let rs = creg2reg_idx(rsc);
+
+  let rsext = X(rs) + immext;
+  X(rs) = rsext;
+  
+   match execute(LOAD(imm, rs, rd, false, WORD, false, false)) {
+     RETIRE_SUCCESS => {
+                         X(rd) = EXTZ(X(rd)[15..0]);
+                         RETIRE_SUCCESS
+                       },
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
+}
+
+mapping clause assembly = C_LHU(uimm, rsc, rdc)
+  <-> "c.lhu" ^ spc() ^ creg_name(rdc) ^ sep() ^ creg_name(rsc) ^ sep() ^ hex_bits_4(uimm @ 0b000)
+/* ****************************************************************** */
+union clause ast = C_LH : (bits(1), cregidx, cregidx)
+
+//$[wavedrom FUNCT3 _ _ _ _ _ C0]
+mapping clause encdec_compressed = C_LH(ui5, rs1, rd)
+  <-> 0b100 @ 0b001 @ rs1 : cregidx @ 0b1 @ ui5 : bits(1) @ rd : cregidx @ 0b00
+
+function clause execute (C_LH(uimm, rsc, rdc)) = {
+  let immext : xlenbits = EXTZ(uimm);
+  let imm : bits(12) = EXTZ(uimm);
+  
+  let rd = creg2reg_idx(rdc);
+  let rs = creg2reg_idx(rsc);
+
+  let rsext = X(rs) + immext;
+  X(rs) = rsext;
+  
+   match execute(LOAD(imm, rs, rd, false, WORD, false, false)) {
+     RETIRE_SUCCESS => {
+                         X(rd) = EXTS(X(rd)[15..0]);
+                         RETIRE_SUCCESS
+                       },
+     RETIRE_FAIL => RETIRE_FAIL,
+   }
+}
+
+mapping clause assembly = C_LH(uimm, rsc, rdc)
+  <-> "c.lh" ^ spc() ^ creg_name(rdc) ^ sep() ^ creg_name(rsc) ^ sep() ^ hex_bits_4(uimm @ 0b000)
+/* ****************************************************************** */
+
+union clause ast = C_SB : (bits(2), cregidx, cregidx)
+
+//$[wavedrom FUNCT3 _ _ _ _ C0]
+mapping clause encdec_compressed = C_SB(ui65, rs1, rs2)
+  <-> 0b100 @ 0b010 @ rs1 : cregidx @ ui65 : bits(2) @ rs2 : cregidx @ 0b00
+
+function clause execute (C_SB(uimm, rs1c, rs2c)) = {
+  let immext : xlenbits = EXTZ(uimm);
+  let imm : bits(12) = EXTZ(uimm);
+
+  let rs1c = creg2reg_idx(rs1c);
+  let rs2c = creg2reg_idx(rs2c);
+
+  let rsext = X(rs1c) + immext;
+  
+  let rs2c_lbyte = X(rs2c)[7..0];
+
+  X(rs1c) = rsext;
+  //X(rs2c) = EXTZ(rs2c_lbyte);
+  
+  X(rs2c) = EXTZ(X(rs2c)[7..0]);
+  //mem(rsext[7..0]) = X(rs2c);
+  
+  match execute(STORE(imm, rs2c, rs1c, WORD, false, false)){
```

Comment:
WORD instead of BYTE ? and the match statements feels redundant ?  rs2c_lbyte declared but not used  ?  Comments redundant ? I dont understand the point of rsext ?

Reply:
Yes, exactly, one of many implementations in this PR full of nonsense. With the way the current PR stands, finding and pointing out all the problems would likely take longer than just reimplementing it properly from scratch, so offering a full code review is a waste of reviewers' time.

Reply:
If i do not add the match and just do an execute, then Sail throws an error, so the match was used.
Regarding the commented out line, that was the previous implementation that i had done, but, now is done with the STORE.

I did look for the BYTE Store, looks like i did not look for it properly.